File: collection/view/addCollection.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-

from django.shortcuts import render
from django.http import JsonResponse
from collection.models import collection

def goToAddCollection(request):
    context = {}
    idBook = request.GET["idBook"]

    if "readerId" not in request.session:
        lastUrl = request.GET["lastUrl"]

        context['status'] = "success1"
        context['message'] = "/reader/login/?lastUrl=" + "/collection/addCollectionInter/?idBook=" + idBook
        return JsonResponse(context)
    else:
        readerId = request.session["readerId"]

        context['status'] = "success2"
        context['message'] = idBook
        # context['message'] = "/collection/add/?idBook=" + idBook
        return JsonResponse(context)

def add(request):
    if "readerId" not in request.session:
        return render(request, 'reader/login.html')

    idBook = request.GET["idBook"]
    idReader = request.session["readerId"]
    context = {}
    try:
        res, status, mes = collection.add(idReader, idBook)
        # res, status, mes = collection.add(idBook, idReader)

        if not res:
            context['status'] = "fail"
            context['errorNumber'] = status
            context['message'] = "錯誤 : add　寫入 collection 表錯誤。"
            return JsonResponse(context)

        context['status'] = "success"
        context['errorNumber'] = 170400
        context['message'] = "添加成功．"
        return JsonResponse(context)
    except Exception as e:
        context['status'] = "fail"
        context['errorNumber'] = 170401
        context['message'] = str(e)
        print str(e)
        return JsonResponse(context)

File: djangoStatic/templatesJs/collection/addCollection.js
function addCollection(idBookArg) {
  $.ajax({
    url: '/collection/addCollection/',
    type: 'GET',
    dataType: 'json',
    data: {"idBook": idBookArg}
  })
  .done(function(resp) {
    // console.log(resp);
    var errorNumber = resp.errorNumber;
    var message = resp.message;
    if (resp.status == "success" && message != null) {
      alert("收藏成功！");
      window.location.href = "/chapter/bookChapter/?idBook=" + idBookArg
    }
    else if (resp.status == "fail") {
      alert(errorNumber + "收藏失敗："　+ message)
    }
    else{
      alert("未知錯誤：　" + errorNumber　+ message)
    }
  })
  .fail(function(resp) {
    console.log("addCollection fail");
    console.log(resp);
  });
}

function goToAddCollection() {
  var lastUrl = window.location.href;
  var idBook = $("#idBook").val();

  $.ajax({
    url: '/collection/goToAddCollection/',
    type: 'GET',
    dataType: 'json',
    data: {"idBook": idBook, "lastUrl": lastUrl}
  })
  .done(function(resp) {
    // console.log(resp);
    if (resp.status == "success1") {
      window.location.href = resp.message
    }
    else if (resp.status == "success2") {
      addCollection(resp.message);
    }
    else if (resp.status == "fail") {
      alert(resp.message);
    }
    else{
      alert(resp.message);
    }
  })
  .fail(function(resp) {
    console.log(resp);
  });
}
